.github/workflows/code-coverage.yml

```diff
@@ -0,0 +1,66 @@
+name: Code Coverage Report
+
+on: [push, pull_request]
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Setup cmake
+      uses: jwlawson/actions-setup-cmake@v2
+      with:
+        cmake-version: '3.22.x'
+    - name: Configure CMake
+      run: cmake -B cmake-build-posix -S executables/referenceApp -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
+
+    - name: Build
+      run: cmake --build cmake-build-posix --target app.referenceApp -j --config ${{env.BUILD_TYPE}}
+
+    - name: Test
+      # Execute tests defined by the CMake configuration.
+      run: |
+          cmake -B cmake-build-unit-tests -S executables/unitTest -DBUILD_UNIT_TESTS=ON
+          cmake --build cmake-build-unit-tests -j4
+          ctest --test-dir cmake-build-unit-tests -j4
+    
+    #Code coverage 
+    - name: Install lcov
+      run: sudo apt install lcov
+    
+    - name: Capture code coverage
+      run: |
+        echo "Capturing code coverage..."
+        lcov --capture --directory . \
+           --output-file cmake-build-unit-tests/coverage_unfiltered.info
+
+    - name: Filter out 3rd party and mock files
+      run: |
+        echo "Filtering out 3rd party and mock files from coverage data..."
+        lcov --remove cmake-build-unit-tests/coverage_unfiltered.info \
+             '*libs/3rdparty/googletest/*' \
+             '*/mock/*' \
+             '*/gmock/*' \
+             '*/usr/include/*' \
+             '*/gtest/*' \
+             --output-file cmake-build-unit-tests/coverage.info
```

Comment:
Yes Oliver, I have excluded the mock/test patterns but I also had to exclude 3rd party files because they were not getting excluded by using --no-external command in the latest commit. It works for excluding system libraries though